### PR TITLE
CSPWFRT: Add post-survey

### PR DESCRIPTION
**Pull request by molly-moen**
[PLC-864](https://codedotorg.atlassian.net/browse/PLC-864)
Add CSP Workshop for Returning Teachers post-survey, created using Foorm.
The new survey is defined in dashboard/config/foorm/forms/surveys/pd/csp_wfrt_post_survey.0.json. This is based on the post-survey defined by the [spec](https://docs.google.com/document/d/1u1Vfm5DnsBpwUrV2XLp9XMcF2R3SU7b1EAtlBBB6wEU/edit#). After this PR goes in we can share a preview easily with the relevant stakeholders and make any requested updates.
This PR also includes the code changes required to give teachers access to this survey and to give facilitators/organizers access to the results:
- survey link will now be sent in the exit mailer
- survey link will be available on /my-professional-learning if the workshop is ended and the user has not filled it out yet
- workshop dashboard will link to appropriate survey results page

This also includes a little refactoring so the frontend has a single source of truth for which workshops should show a foorm results page.

## Links
- [spec](https://docs.google.com/document/d/1u1Vfm5DnsBpwUrV2XLp9XMcF2R3SU7b1EAtlBBB6wEU/edit#)
- [jira](https://codedotorg.atlassian.net/browse/PLC-864)

## Testing story
Tested manually by following a similar process to the one described in [this pr](https://github.com/code-dot-org/code-dot-org/pull/34335), only with a csp wfrt workshop. I validated that the survey matches the spec and the results are summarized correctly on the results page. Also checked the csp wfrt mailer now links to the correct survey and added a unit test for the exit survey url.

# Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/code-studio/pd/workshop_dashboard/workshop_summary_utils.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import {SubjectNames} from '@cdo/apps/generated/pd/sharedWorkshopConstants';
+
+export function useFoormSurvey(subject, lastSessionDate) {
+  // Local summer, CSF Intro, or CSP for returning teachings workshop
```
Comment (breville) [inline apps/src/code-studio/pd/workshop_dashboard/workshop_summary_utils.js]: Is "returning teachings workshop" the same as "WFRT" elsewhere?

File: dashboard/app/models/pd/workshop.rb [hunk 2]
```diff
@@ -679,6 +679,10 @@ def local_summer?
     subject == SUBJECT_SUMMER_WORKSHOP
   end
 
+  def csp_wfrt?
```
Comment (breville) [inline dashboard/app/models/pd/workshop.rb:683]: Can we have a comment defining this acronym?

File: apps/src/code-studio/pd/workshop_dashboard/workshop_summary_utils.js (hunk 1 above)
Comment (molly-moen) [reply apps/src/code-studio/pd/workshop_dashboard/workshop_summary_utils.js]: yeah wfrt stands for "workshop for returning teachers"

File: dashboard/app/models/pd/workshop.rb (hunk 2 above)
Comment (molly-moen) [reply dashboard/app/models/pd/workshop.rb:683]: will do

**Review by breville — APPROVED** [review]
I think we could unify a bit on terminology on the new course, but otherwise looks great.  Centralising the truth is also a nice improvement!

File: apps/src/code-studio/pd/workshop_dashboard/workshop_summary_utils.js (hunk 1 above)
Comment (molly-moen) [reply apps/src/code-studio/pd/workshop_dashboard/workshop_summary_utils.js]: realized it should have said teachers, not teachings, fixed

**Pull request merged**